src/main/java/com/awesomeorg/airlineservice/repository/TicketRepository.java

```diff
@@ -1,9 +1,20 @@
 package com.awesomeorg.airlineservice.repository;
 
 import com.awesomeorg.airlineservice.entity.Ticket;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
+import java.time.LocalDate;
+
 @Repository
 public interface TicketRepository extends JpaRepository<Ticket, Long> {
+    @Query("SELECT t FROM tickets t " +
+            "LEFT JOIN reservations r ON t.id = r.ticketId AND r.reservationDate = :date " +
+            "WHERE r.id IS NULL")
+    Page<Ticket> findFreeTicket(@Param("date") LocalDate reservationDate, PageRequest pageRequest);
```

Comment:
I think here is some issue with the logic. That means, that only 1 ticket can be purchased for the same reservation? It worked for my example, where one table couldn't be booked by multiple people at the same time but not applicable for the airline. Seems like we have some flaws here with the model. Looking again at the [diagram](https://github.com/VeronicaZhigalova/airline-service/blob/52bd8ab1cee4902b29f53250beb4b8581c3fcab4/data_model.png) , is the fk_seat_id the id in the ticket?. In theory, it could be simplified or made more logical.

Reservation and Ticket is pretty much the same. Maybe they should be combined. Then, baggage relates to the Reservation. 

 private String departureAirport;

    private String arrivalAirport;

    @JsonFormat(pattern = "yyyy-MM-dd")
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private LocalDate departureTime;

    @JsonFormat(pattern = "yyyy-MM-dd")
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private LocalDate arrivalTime;

    @JsonFormat(pattern = "yyyy-MM-dd")
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private LocalDate departureDate;

    @JsonFormat(pattern = "yyyy-MM-dd")
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private LocalDate returnDate;
    
    private String departure;

    private String destination;
    
Could be moved to the Flight class that is referenced in the Reservation. It would be two fields, one for the inbound flight and one for the outbound flight